packages/block-editor/src/components/block-tools/style.scss

```diff
@@ -119,6 +118,169 @@
+					margin-bottom: $grid-unit + $grid-unit-05;
+					background-color: $gray-300;
+					position: absolute;
+					left: 44px;
+					top: -1px;
+				}
+			}
+
+			& > .block-editor-block-toolbar__group-expand-fixed-toolbar {
+				border: none;
+
+				// Add a border as separator in the block toolbar.
+				&::before {
+					content: "";
+					width: $border-width;
+					margin-top: $grid-unit + $grid-unit-05;
+					margin-bottom: $grid-unit + $grid-unit-05;
+					background-color: $gray-300;
+					position: relative;
+					left: -12px; //the padding of buttons
+					height: 24px;
+				}
+			}
+
+			.show-icon-labels & {
+				left: $grid-unit-80 + $grid-unit-50;
+
+				&.is-collapsed {
+					left: $grid-unit-10 * 56;
+				}
+
+				.is-fullscreen-mode & {
+					left: $grid-unit-80 + $grid-unit-80;
+				}
+
+				.block-editor-block-parent-selector .block-editor-block-parent-selector__button::after {
+					left: 0;
+				}
+
+				.block-editor-block-toolbar__block-controls .block-editor-block-mover {
+					border-left: none;
+					&::before {
+						content: "";
+						width: $border-width;
+						margin-top: $grid-unit + $grid-unit-05;
+						margin-bottom: $grid-unit + $grid-unit-05;
+						background-color: $gray-300;
+						position: relative;
+					}
+				}
+			}
+		}
+
+		&.is-fixed .block-editor-block-parent-selector {
+			.block-editor-block-parent-selector__button {
+				position: relative;
+				top: -1px;
+				border: 0;
+				padding-right: 6px;
+				padding-left: 6px;
+				&::after {
+					content: "\00B7";
+					font-size: 16px;
+					line-height: $grid-unit-40 + $grid-unit-10;
+					position: absolute;
+					left: $grid-unit-40 + $grid-unit-15 + 2px;
+					bottom: $grid-unit-05;
+				}
+			}
+		}
+
+		&:not(.is-fixed) .block-editor-block-parent-selector {
+			position: absolute;
+			top: -$border-width;
+			left: calc(-#{$grid-unit-60} - #{$grid-unit-10} - #{$border-width});
+
+			.show-icon-labels & {
+				position: relative;
+				left: auto;
+				top: auto;
+				margin-top: -$border-width;
+				margin-left: -$border-width;
+				margin-bottom: -$border-width;
+			}
+		}
+	}
+
+	// on tablet vewports the toolbar is fixed
+	// on top of interface header and covers the whole header
+	// except for the inserter on the left
+	@include break-medium() {
+		&.is-fixed {
+			width: calc(100% - #{$grid-unit-80 + $grid-unit-40});
+			&.is-collapsed {
+				// when collapsed minimize area
+				width: initial;
+			}
+			.is-fullscreen-mode & {
+				width: calc(100% - #{$grid-unit-80 + $grid-unit-70});
+				&.is-collapsed {
+					// when collapsed minimize area
+					width: initial;
+				}
+			}
+		}
+	}
+
+	// on desktop viewports the toolbar is fixed
+	// on top of interface header and leaves room
+	// for the block inserter the publish button
+	@include break-large() {
+		&.is-fixed {
+			// leave room for block inserter and the dashboard navigation
+			left: $grid-unit-80 + $grid-unit-40 + ( $grid-unit-80 * 2 );
```

Comment:
I had to add an extra 160 grid units to account for the dashboard navigation on larger viewports. @draganescu does this seem reasonable?

Reply:
Ha yet more CSS fiddling. I think I never opened the sidebar when not in full screen 😱

Reply:
It's not OK because it now breaks when the sidebar is collapsed. I hope there is a body class for that!